src/wymeditor/core.js

```diff
@@ -541,251 +831,39 @@ jQuery.extend(WYMeditor, {
 
     `jQuery(".wymeditor").wymeditor({});`
 */
-jQuery.fn.wymeditor = function (options) {
-    var $textareas = this;
-
-    options = jQuery.extend({
-
-        html:       "",
-        basePath:   false,
-        wymPath:    false,
-        iframeBasePath: false,
-        jQueryPath: false,
-        skin:       "default",
-        lang:       "en",
-        direction:  "ltr",
-        customCommands: [],
-
-        // These values will override the default for their respective
-        // `DocumentStructureManager.structureRules`
-        structureRules: {
-            defaultRootContainer:  'p'
-        },
+jQuery.fn.wymeditor = function (providedOptions) {
+    var $textareas = this,
+        mergedOptions,
+        optionsWithDefaults,
+        optionsWithSkinOverrides,
+        skin;
+
+
+    optionsWithDefaults = jQuery.extend({}, WYMeditor.DEFAULT_OPTIONS, providedOptions);
```

Comment:
I wrote it like this on purpose because it makes it very explicit what each `extend` is adding and how the precedence works. I think it provides more clarity than a single merge.